ui-modules/blueprint-composer/app/components/util/d3-blueprint.js

```diff
@@ -774,18 +774,38 @@ export function D3Blueprint(container, options) {
     function drawRelationships() {
         showRelationships();
 
+        // Generates unique relation ID string.
+        const getRelationId = (relationDataItem) => (relationDataItem.source._id + '_related_to_' + relationDataItem.target._id);
+
+        // Group relationships per direction, as a key.
+        let relationshipsPerDirection = new Map();
+        _d3DataHolder.visible.relationships.forEach(relation => {
+            const directionKey = getRelationId(relation);
+            if (!relationshipsPerDirection.has(directionKey)) {
+                relationshipsPerDirection.set(directionKey, []);
+            }
+            relationshipsPerDirection.get(directionKey).push(relation.label);
+        });
+
+        // Calculates offset for label, based on number of labels in the same direction.
+        const getLabelOffset = (relationDataItem) => {
+            const directionKey = getRelationId(relationDataItem);
+            let labelIndex = relationshipsPerDirection.get(directionKey).findIndex(label => label === relationDataItem.label);
```

Comment:
Since the arrays have string elements we can simply used `.indexOf` here like
`let labelIndex = relationshipsPerDirection.get(directionKey).indexOf(relationDataItem.label);`

Reply:
Improved in 8fa7e4decb16d1a1abd7eb93bd0bae088f44023b